File: backend/components/metiers/administration_publique/utils/views/admin/admin_views.test.py
"""
admin_views.test.py – Tests unitaires et conformité admin views threed (Python)
"""
from fastapi.testclient import TestClient
from .admin_views import router
from fastapi import FastAPI

app = FastAPI()
app.include_router(router)
client = TestClient(app)

def test_admin_action():
    response = client.post("/admin/action", json={"action": "activate", "user": "admin", "details": "test"})
    assert response.status_code == 200
    data = response.json()
    assert data["action"] == "activate"
    assert data["user"] == "admin"
    assert data["status"] == "success"
    # Test action non supportée
    response2 = client.post("/admin/action", json={"action": "hack", "user": "admin"})
    assert response2.status_code == 400

File: backend/components/metiers/administration_publique/utils/views/admin/admin_views.test.js
// admin_views.test.js – Tests unitaires et conformité admin views threed (JS)
const request = require('supertest');
const express = require('express');
const adminRouter = require('./admin_views');

describe('Admin Threed', () => {
  const app = express();
  app.use(express.json());
  app.use('/admin', adminRouter);

  it('doit effectuer une action admin supportée', async () => {
    const res = await request(app)
      .post('/admin/action')
      .send({ action: 'activate', user: 'admin', details: 'test' });
    expect(res.statusCode).toBe(200);
    expect(res.body.action).toBe('activate');
    expect(res.body.user).toBe('admin');
    expect(res.body.status).toBe('success');
  });

  it('refuse une action non supportée', async () => {
    const res = await request(app)
      .post('/admin/action')
      .send({ action: 'hack', user: 'admin' });
    expect(res.statusCode).toBe(400);
  });
});
